pkg/kubectl/cmd/util/factory.go

```diff
@@ -106,8 +108,23 @@ func NewFactory(optionalClientConfig clientcmd.ClientConfig) *Factory {
 		clientConfig = DefaultClientConfig(flags)
 	}
 
-	clients := NewClientCache(clientConfig)
+	clients := newClientCache(clientConfig)
 
+	var experimentalClient *client.ExperimentalClient
+	cfg, experimentalClientErr := clientConfig.ClientConfig()
```

Comment:
Factory creation is successful regardless of whether this succeeds. The error is returned when the user asks for a experimental client. I also don't see anywhere where we manipulate the clientConfig so this should have the same behavior.